documentation/api/query/v4/operators.markdown

```diff
@@ -299,19 +299,59 @@ An `in` statement constitutes a full query string, which can be used alone or as
 
 "In" statements are **non-transitive** and take two arguments:
 
-* The first argument **must** consist of one or more **fields** for the endpoint or entity **being queried.**. This is a string or vector of strings.
+* The first argument **must** consist of one or more **fields** for the endpoint
+  or entity **being queried.**. This is a string or vector of strings.
 * The second argument **must** be either:
-** an **`extract` statement,** which acts as a list of fields to extract during the subquery for matching against the **fields** in the `in` clause.
-** a **`from` statement,** which sets the context, and allows for an extract statement to be provided. *Note:* this syntax is new and experimental.
+** an **`extract` statement,** which acts as a list of fields to extract during
+   the subquery for matching against the **fields** in the `in` clause.
+** a **`from` statement,** which sets the context, and allows for an extract
+   statement to be provided. *Note:* this syntax is new and experimental.
+** an **`array` statement,** which acts as a list of values to match against the
+   **field** in the `in` clause.
 
 **Matches if:** the field values are included in the list of values created by the `extract` or `from` statement.
 
+##### `array`
+
+An `in` statement also accepts an `array` statement as a second argument.
+
+"Array" statements take a single vector argument of values to match the first
+argument of `in` against.
+
+The following query filters for the nodes, `foo.local`, `bar.local`, and
+`baz.local`:
+
+    ["in", "certname",
+     ["array",
```

Comment:
So I think there's some precedence for the literal decorator with `["fact", "..."]`. Also I think without the decorator could cause problems, i.e. `[in ["fact" "name"] ["extract", ...` becomes ambiguous, is `extract` a keyword or is it a member of an array literal? we could use `["array" &rest]`, such as `[array foo bar`] which means the literal `[foo bar]`, sort of like the clojure vector constructor, but then this means a `map` would start to look like `[map [foo baz] [bar buzz]`, instead of `{foo baz ...}`. I thought of `array` and `fact`, as casting operators, converting something that looks ambiguous. What would be your concerns with a literal declarator?